sql/moz-fx-data-shared-prod/contextual_services_derived/event_aggregates_v1/query.sql

```diff
@@ -172,20 +159,68 @@ WITH combined AS (
     CAST(
       (SELECT value FROM UNNEST(events[SAFE_OFFSET(0)].extra) WHERE key = 'position') AS INT64
     ) AS position,
-    CASE
-    WHEN
-      metrics.url2.top_sites_contile_reporting_url IS NULL
-    THEN
-      'remote settings'
-    ELSE
-      'contile'
-    END
+    -- Only Contile is available for mobile tiles.
+    'contile' AS provider,
     AS provider,
     -- `match_type` is only available for `quicksuggest_*` tables
     NULL AS match_type,
     normalized_os,
   FROM
     org_mozilla_fenix.topsites_impression
+  UNION ALL
+  SELECT
+    -- Due to the renaming (from 'topsite' to 'topsites'), some legacy Firefox
+    -- versions are still using the `topsite` key in the telemetry
+    IFNULL(metrics.uuid.top_sites_context_id, metrics.uuid.top_site_context_id) AS context_id,
+    DATE(submission_timestamp) AS submission_date,
+    'topsites' AS source,
+    IF(events[SAFE_OFFSET(0)].name = 'contile_click', 'click', 'impression') AS event_type,
+    -- The adMarketplace APIs accept form factors out of "desktop", "phone", or "tablet";
+    -- we are currently always using "phone" for Fenix, so stay consistent with that here.
+    'phone' AS form_factor,
+    normalized_country_code AS country,
+    metadata.geo.subdivision1 AS subdivision1,
+    IFNULL(metrics.string.top_sites_contile_advertiser, metrics.string.top_site_contile_advertiser) AS advertiser,
+    'release' AS release_channel,
+    CAST(
```

Comment:
do the `CAST` calls in here need to be `SAFE_CAST` to guard against non int values?

Reply:
I'd agree it's generally a good idea to prefer `SAFE_CAST`